clang/test/Frontend/sycl-propagate-aspect-warning.cpp

```diff
@@ -5,50 +5,50 @@
 
 #include "sycl.hpp"
 
-// expected-note@+1 4 {{propagated from call to function 'Struct1::Struct1()'}}
+// expected-note-re@+1 4 {{propagated from call to function '{{.*}}Struct1::Struct1({{.*}})'}}
```

Comment:
Is `re` required to treat {{.*}} as a regular expression? I can't recall but doesn't `expected-note` parse this as 0 or more?

Reply:
> Is `re` required to treat {{.*}} as a regular expression?

Without the `-re` post-fix it will look for `{{.*}}` as part of the expected note.

> I can't recall but doesn't expected-note parse this as 0 or more?

What do you mean? The regexes mean "zero-or-more occurances of any character" which we need here because in Linux testing we get `Struct1::Struct1()` while on Windows we get `public: __cdecl Struct1::Struct1(void)`, so `{{.*}}Struct1::Struct1({{.*}})` matches the common parts.

Reply:
I got confused between CHECK and 'expected' directives and I thought the `'-re'` wasn't required here. My bad.